ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -208,7 +212,12 @@ public NodeRecord updateNodeRecord() {
                       nodeRecordFactory.createFromValues(
                           sequenceNumber,
                           new EnrField(EnrField.ID, IdentitySchema.V4),
-                          new EnrField(EnrField.PKEY_SECP256K1, Functions.compressPublicKey(id)),
+                          new EnrField(
```

Comment:
I'm not 100% sure if we should change the key from `secp256k1`. I understand that it will be weird to have the key saying `k1` and the actual value is an `r1` public key. However, the ENR spec defined the key name to be `secp256k1` (unfortunately).

Maybe for now we leave the key as it was. I'll follow up and see if is is possible to update the spec.

For reference: https://eips.ethereum.org/EIPS/eip-778

Reply:
@RatanRSur it would be great to hear your thoughts on this.

Reply:
After testing it: The `compressPublicKey` function needs to use the same elliptic curve as the node id, if not an error is caused. We decided to keep the key name as `EnrField.PKEY_SECP256K1` to be in compliance with the standard even though the value could be derived with another elliptic curve.

Reply:
What was done before with getting the name of the curve is fine for the mainnet side as it uses the same value as the key.

On the NIST side, it changes this EIP non-functionally. However, strict adherence to the EIP is not really required as changing the cryptography already breaks mainnet compatibility. I would probably allow the name to change since it's less confusing for future people in NIST-land and adds some incompatibility to the ENRs. I don't think it's likely node operators will accidentally pollute themselves with NIST ENRs gotten by extra-protocol means but it's possible.